c7n/resources/ram.py

```diff
@@ -0,0 +1,34 @@
+from c7n.actions import ActionRegistry
+from c7n.filters import FilterRegistry
+from c7n.manager import resources
+from c7n.query import QueryResourceManager, TypeInfo, DescribeSource
+from c7n.tags import universal_augment, Tag, RemoveTag
+
+
+class GetResourceShare(DescribeSource):
+    def augment(self, resources):
+        return universal_augment(self.manager, super().augment(resources))
+
+
+filters = FilterRegistry('ram.filters')
+actions = ActionRegistry('ram.actions')
+
+
+@resources.register('ram')
```

Comment:
Since RAM is the service rather than the resource type we're targeting here, it might be clearer to use a more specific name like `ram-share` or `ram-resource-share`.